### PR TITLE
Fix: `RaceConditionTests.isNonConsumingExercise` is really `isExercise` [KVL-1207]

### PR DESCRIPTION
**Pull request by fabiotudone-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/RaceConditionTests.scala [hunk 1]
```diff
@@ -107,7 +107,7 @@ private[testtool] object RaceConditionTests {
       tx.hasEventsNumber(1) &&
         tx.containsEvent(isExerciseEvent(RaceTests.ContractWithKey.ChoiceArchive))
 
-    def isNonConsumingExercise(tx: TransactionTree): Boolean =
+    def isExercise(tx: TransactionTree): Boolean =
```
Comment (cocreature) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/RaceConditionTests.scala:110]: I’m a bit confuse. Afaict we filter by the choice name and the choice is non-consuming https://github.com/digital-asset/daml/blob/800f3741024e176eb9bdc1a1d08c18c01e5bee67/ledger/test-common/src/main/daml/semantic/RaceTests.daml#L26 Why are you suggesting that it’s not non-consuming?

Comment (fabiotudone-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/RaceConditionTests.scala:110]: Oh you're right, but then we have a problem in `RWArchiveVsRollbackConsumingChoice` I believe

Comment (cocreature) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/RaceConditionTests.scala:110]: good catch, will look into it today if you don’t beat me to it.

Comment (cocreature) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/RaceConditionTests.scala:110]: actually I think the problem is much bigger. All the filters in there seem crap :cry:

Comment (fabiotudone-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/RaceConditionTests.scala:110]: Kool

**Comment by fabiotudone-da** [comment]
Closing as this is not the correct fix for the problem.

**Pull request closed without merging**